src/components/views/rooms/BasicMessageComposer.tsx

```diff
@@ -282,9 +282,7 @@ export default class BasicMessageEditor extends React.Component<IProps, IState>
     };
 
     private showPlaceholder(): void {
-        // escape single quotes
-        const placeholder = this.props.placeholder?.replace(/'/g, "\\'");
-        this.editorRef.current?.style.setProperty("--placeholder", `'${placeholder}'`);
+        this.editorRef.current?.style.setProperty("--placeholder", `'${CSS.escape(this.props.placeholder ?? "")}'`);
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/CSS/escape_static says its intended for use for escaping selectors rather than values, are we sure its the right method to use here?

Reply:
Yes, that's it's primary us. The docs you linked also point at the possible use for escaping general strings:

> The `escape()` method can also be used for escaping strings, although it escapes characters that don't strictly need to be escaped

The over-escaping didn't seem to hurt in my testing and I couldn't find anything better / easier.

I think the alternative would be to extend the custom escaping to also cover backslashes? If you think that's better, I could also try that. Just seemed more complicated and error prone at first sight.